woocommerce/woocommerce
# Implement the "Installed extensions" card

### All Submissions:


### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/34025.

This PR adds the "Installed extensions" card to the Multichannel Marketing experience.

Some points to take note of:

- I have made some changes to the directory structure: now we have `overview-multichannel` directory at the same level with the existing `overview` directory, both sitting under the `marketing` directory. This allows both new and existing components to reuse the same wp.data in https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-admin/client/marketing/data.
- I added a new component called `CollapsibleCard` in the shared components directory. This component will be used in subsequent PRs too.
- The new marketing components get their inspirations from the existing components in https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-admin/client/marketing/overview/installed-extensions. The difference is that the new ones use TypeScript, React functional component and hooks, while the existing ones use JavaScript, class components and HOCs.
- The new "Installed extensions" card can expand / collapse. It is expanded by default.
- I use an existing track event name `marketing_installed_options` with a new event prop value `link: 'manage'` for the "Manage" button, see https://github.com/woocommerce/woocommerce/pull/34080/files#diff-41979fbae527b40e579cdcda647b00d99b65fd0404f2577ca187bbf1726f2bb3R67-R70.

Note that the "Installed extensions" card in this PR is different from the one in the figma design (7cNed5RvTasoLgn38Hea0R-fi-2116%3A81074). In Figma, social channel plugins (e.g. Google, Pinterest etc) do not show up in "Installed extensions" card but in the "Channels" card; in this PR, the social channel plugins will show up in "Installed extensions" card. This is so that we can release the features incrementally. When we work on the "Channels" card later, the social channel plugins should be removed from this "Installed extensions" card.

📹 Demo video with my voice:

https://user-images.githubusercontent.com/417342/181312100-54783465-b740-4546-b4e4-ba437c7c288b.mov

### How to test the changes in this Pull Request:

1. Enable the new Marketing experience in `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Go to the Marketing > Overview page in `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`. You should see the "Installed extensions" card.
3. The card should be expanded by default. You should be able to collapse / expand the card by clicking on the toggle button in the top right corner of the card.
4. Each plugin should have a button:
    - Plugins that are installed but not yet activated should have an "Activate" button. Clicking on it will activate the plugin.
    - Plugins that are activated but not yet configured should have a "Finish setup" button. Clicking on it will bring you to the settings URL.
    - Plugins that are activated and configured should have a "Manage" button. Clicking on it will bring you to the plugin dashboard URL or settings URL.
    - All the buttons above should fire track events. The track events are similar to those in the existing https://github.com/woocommerce/woocommerce/blob/30de9ccb1917e6fd7a5dfbc754217b8f603c811a/plugins/woocommerce-admin/client/marketing/overview/installed-extensions/row.js file.

### Other information:


### FOR PR REVIEWER ONLY:
